### PR TITLE
[CBR-322] Tests for block metadata

### PR DESCRIPTION
**Pull request by uroboros**
## Description

Tests for BlockMeta data

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CBR-322

## Type of change

- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [x] 🔨 New or improved tests for existing code

## Testing checklist

- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: wallet-new/test/unit/Test/Spec/Kernel.hs [hunk 1]
```diff
@@ -67,17 +80,66 @@ spec =
...
 
+    -- | Translates the DSL BlockMeta' value to BlockMeta
+    intBlockMeta :: forall h. (Hash h Addr, Interpret h (BlockMeta' h))
+                 => IntCtxt h
+                 -> (BlockMeta' h)
+                 -> TranslateT IntException IO BlockMeta
+    intBlockMeta intCtxt a = do
+        ma' <- catchTranslateErrors $ runIntT' intCtxt $ int a
+        case ma' of
+          Left err -> liftIO $ throwM err
```
Comment (parsonsmatt) [inline wallet-new/test/unit/Test/Spec/Kernel.hs]: Is the intent to throw the exception in `IO`? It seems like `TranslateT` has some means of carrying exception/error information.

Comment (edsko) [reply wallet-new/test/unit/Test/Spec/Kernel.hs]: I guess the point is to have the interpretation exception be thrown up to the test framework, and have the test marked as failed should that happen.

Comment (parsonsmatt) [reply wallet-new/test/unit/Test/Spec/Kernel.hs]: Sounds good :)

**Comment by edsko** [comment]
Replaced by https://github.com/input-output-hk/cardano-sl/pull/3413.

**Pull request closed without merging**